src/plugins/discover/public/components/discover_tour/discover_tour_provider.tsx

```diff
@@ -0,0 +1,306 @@
+import { useDiscoverServices } from '../../utils/use_discover_services';
+import { DiscoverTourContext, DiscoverTourContextProps } from './discover_tour_context';
+import { DISCOVER_TOUR_STEP_ANCHORS } from './discover_tour_anchors';
+
+const MAX_WIDTH = 350;
+
+interface TourStepDefinition {
+  anchor: EuiTourStepProps['anchor'];
+  title: EuiTourStepProps['title'];
+  content: EuiTourStepProps['content'];
+  imageName: string;
+  imageAltText: string;
+}
+
+const tourStepDefinitions: TourStepDefinition[] = [
+  {
+    anchor: DISCOVER_TOUR_STEP_ANCHORS.addFields,
+    title: i18n.translate('discover.dscTour.stepAddFields.title', {
+      defaultMessage: 'Add fields to the table',
+    }),
+    content: (
+      <FormattedMessage
+        id="discover.dscTour.stepAddFields.description"
+        defaultMessage="Click {plusIcon} to add the fields that interest you."
+        values={{
+          plusIcon: <EuiIcon size="s" type="plusInCircleFilled" color="primary" aria-label="+" />,
+        }}
+      />
+    ),
+    imageName: 'add_fields.gif',
+    imageAltText: i18n.translate('discover.dscTour.stepAddFields.imageAltText', {
+      defaultMessage:
+        'In the Available fields list, click the plus icon to toggle a field into the document table.',
+    }),
+  },
+  {
+    anchor: DISCOVER_TOUR_STEP_ANCHORS.reorderColumns,
+    title: i18n.translate('discover.dscTour.stepReorderColumns.title', {
+      defaultMessage: 'Order the table columns',
+    }),
+    content: (
+      <FormattedMessage
+        id="discover.dscTour.stepReorderColumns.description"
+        defaultMessage="Drag columns to the order you want."
+      />
+    ),
+    imageName: 'reorder_columns.gif',
+    imageAltText: i18n.translate('discover.dscTour.stepReorderColumns.imageAltText', {
+      defaultMessage: 'Use the Columns pop-up to drag the columns to the order you prefer.',
+    }),
+  },
+  {
+    anchor: DISCOVER_TOUR_STEP_ANCHORS.sort,
+    title: i18n.translate('discover.dscTour.stepSort.title', {
+      defaultMessage: 'Sort on one or more fields',
+    }),
+    content: (
+      <FormattedMessage
+        id="discover.dscTour.stepSort.description"
+        defaultMessage="Use the column heading to sort on a single field, or the popover for multiple fields."
+      />
+    ),
+    imageName: 'sort.gif',
+    imageAltText: i18n.translate('discover.dscTour.stepSort.imageAltText', {
+      defaultMessage:
+        'Click a column header and select the desired sort order. Adjust a multi-field sort using the fields sorted pop-up.',
+    }),
+  },
+  {
+    anchor: DISCOVER_TOUR_STEP_ANCHORS.changeRowHeight,
+    title: i18n.translate('discover.dscTour.stepChangeRowHeight.title', {
+      defaultMessage: 'Change the row height',
+    }),
+    content: (
+      <FormattedMessage
+        id="discover.dscTour.stepChangeRowHeight.description"
+        defaultMessage="Adjust the number of lines to fit the contents."
+      />
+    ),
+    imageName: 'rows_per_line.gif',
```

Comment:
Very tiny nit - obviously, might not be possible to make any changes to this at this point.

The style of the fields/document rows is not consistent between gifs, which is especially obvious when comparing the `rows_per_line.gif` to the `sort.gif` - they're supposed to both be showing the same view of the document explorer and simply highlighting different buttons at the top, but they switch styles from rounded/softer to sharp corners.

Reply:
@Heenawter please always give us all the feedback you've got! much appreciated. I updated it to this one (cc @jughosta)
![sortv2](https://user-images.githubusercontent.com/4016496/167909623-c07c99c8-58f7-4757-b7c5-0b3c8c18b98c.gif)

Reply:
@andreadelrio Would it be possible to add some space to the left from the sorting popover so it's not so close to the left side of the gif? 

<img width="513" alt="Screenshot 2022-05-12 at 10 13 02" src="https://user-images.githubusercontent.com/1415710/168024174-7624c6ee-3069-4c79-8859-f78054793b64.png">

Reply:
@jughosta how about with this?
![sortv3cp](https://user-images.githubusercontent.com/4016496/168144828-6447156c-b1ab-4c3b-bc82-4a9380921bb3.gif)

Reply:
@andreadelrio Thanks!